### DISCUSSION TITLE
Autocomplete not showing all the options

### DISCUSSION THREAD

**Discussion by stoplion** [mantinedev/mantine#1330] [Q&A] [answered]
Hello,
I am trying to populate the autocomplete with a large list.

<img width="304" alt="image" src="https://user-images.githubusercontent.com/231343/165199860-ab23180e-5e28-4508-abe0-11def77f54ab.png">

It only shows the first 4 options alphabetically. There should be more than 4 options shown in the dropdown. 

I'm not adding any limits to the Autocomplete:

```
<Autocomplete
            required
            label='Ticker'
            placeholder='BTC'
            data={tickers}
            {...form.getInputProps('ticker')}
          />
```

If I add an arbitrary number to show..

```
<Autocomplete
            required
            label='Ticker'
            placeholder='BTC'
            data={tickers}
            limit={24}
            {...form.getInputProps('ticker')}
          />
```

It shows more options. 

<img width="318" alt="image" src="https://user-images.githubusercontent.com/231343/165200154-24310f1c-4f93-4d2b-aa35-b15b6d723c35.png">

Is there a way to show ALL the options, without having to set a hard coded limit?

Also, is there a way to show the dropdown options in a scrollable box?

**Comment by rtivital** [comment, marked as answer]
No, Autocomplete does not support showing all options. It is not what the component was designed for – you won't expect google search to show you all possible search results in autocomplete, right? If you need all options to be displayed at once, use Select component instead.

**Comment by stoplion** [comment]
Ok. I think limiting the results make sense.

<img width="325" alt="image" src="https://user-images.githubusercontent.com/231343/165371878-2799d47d-207a-4dd2-bcdd-08a36cad4100.png">

Going with Google example, I'd expect the exact match to float to the top.

**Reply by rtivital** [reply]
You can provide your own `filter` function that will sort your data, by default it is displayed in the order that you provide via `data` prop

**Reply by wgpsutherland** [reply]
@rtivital could you expand on what you mean by the filter function allowing you to sort your data? I can't see this anywhere in the docs and it is functionality i've been looking for!

**Reply by rtivital** [reply]
`<Autocomplete filter={yourFunction} />`, it is just a function that returns filtered data based on search query, nothing much to it. You can perform any operations in it.

**Reply by wgpsutherland** [reply]
I thought the interface for the filter function was `filter(value: string, item: AutocompleteItem): boolean;`, where you get one item at a time and simply return true or false: https://github.com/mantinedev/mantine/blob/0ab9ab9321b9df4ff793e142211508211537946a/src/mantine-core/src/Select/Select.tsx#L30
`filterData` seems to be what you're saying, but you can't pass your own to the Autocomplete component:
https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Autocomplete/filter-data/filter-data.ts#L7

**Reply by wgpsutherland** [reply]
@rtivital is my understanding of the above correct? If so, would you be open to accepting a PR for adding a `sort` function prop to the Autocomplete and Select components?

**Reply by ansonkao** [reply]
I've shared a workaround solution for sorting in #3960
